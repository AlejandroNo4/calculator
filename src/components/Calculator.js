import React from 'react';
import App from './App';
import NavBar from './Nav';

const Calculator = () => (
  <div className="body">
    <NavBar />
    <div className="main-container">
      <h1 className="title">Letâ€™s do some math!</h1>
      <App />
    </div>
  </div>
);

export default Calculator;
