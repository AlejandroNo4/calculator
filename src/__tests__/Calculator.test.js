import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import Calculator from '../components/Calculator';

let componentCalculator;

beforeEach(() => {
  componentCalculator = render(<Calculator />);
  const btn = componentCalculator.getByText('AC');
  fireEvent.click(btn);
});

test('render component to Match Snapshot', () => {
  expect(componentCalculator).toMatchSnapshot();
});

test('Clicking buttons will show them in the display', () => {
  const btnTwo = componentCalculator.getByText('5');
  const btnOperator = componentCalculator.getByText('+');
  fireEvent.click(btnTwo);
  fireEvent.click(btnOperator);
  fireEvent.click(btnTwo);

  const result = componentCalculator.getByText('5+5');
  expect(result).toBeTruthy();
});

test('Pushing equal will show the result of a sum 5+5 operation', () => {
  const btnNum = componentCalculator.getByText('5');
  const btnOperator = componentCalculator.getByText('+');
  const btnEqual = componentCalculator.getByText('=');
  fireEvent.click(btnNum);
  fireEvent.click(btnOperator);
  fireEvent.click(btnNum);
  fireEvent.click(btnEqual);

  const result = componentCalculator.getByText('10');
  expect(result).toBeTruthy();
});

test('Pushing equal will show the result of a substraction 55-3 operation', () => {
  const btnNumOne = componentCalculator.getByText('5');
  const btnNumTwo = componentCalculator.getByText('3');
  const btnOperator = componentCalculator.getByText('-');
  const btnEqual = componentCalculator.getByText('=');
  fireEvent.click(btnNumOne);
  fireEvent.click(btnNumOne);
  fireEvent.click(btnOperator);
  fireEvent.click(btnNumTwo);
  fireEvent.click(btnEqual);

  const result = componentCalculator.getByText('52');
  expect(result).toBeTruthy();
});

test('Pushing equal will show the result of a multiplication 5x3 operation', () => {
  const btnNumOne = componentCalculator.getByText('5');
  const btnNumTwo = componentCalculator.getByText('3');
  const btnOperator = componentCalculator.getByText('X');
  const btnEqual = componentCalculator.getByText('=');
  fireEvent.click(btnNumOne);
  fireEvent.click(btnOperator);
  fireEvent.click(btnNumTwo);
  fireEvent.click(btnEqual);

  const result = componentCalculator.getByText('15');
  expect(result).toBeTruthy();
});

test('Pushing equal will show the result of a division 5รท4 operation', () => {
  const btnNumOne = componentCalculator.getByText('5');
  const btnNumTwo = componentCalculator.getByText('4');
  const btnOperator = componentCalculator.getByText('รท');
  const btnEqual = componentCalculator.getByText('=');
  fireEvent.click(btnNumOne);
  fireEvent.click(btnNumOne);
  fireEvent.click(btnOperator);
  fireEvent.click(btnNumTwo);
  fireEvent.click(btnEqual);

  const result = componentCalculator.getByText('13.75');
  expect(result).toBeTruthy();
});

test('Pushing equal will show the result 10% of 100 operation', () => {
  const btnNumOne = componentCalculator.getByText('1');
  fireEvent.click(btnNumOne);
  const btnNumTwo = componentCalculator.getByText('0');
  const btnOperator = componentCalculator.getByText('%');
  const btnEqual = componentCalculator.getByText('=');
  fireEvent.click(btnNumOne);
  fireEvent.click(btnNumTwo);
  fireEvent.click(btnOperator);
  fireEvent.click(btnNumOne);
  fireEvent.click(btnNumTwo);
  fireEvent.click(btnNumTwo);
  fireEvent.click(btnEqual);

  const result = componentCalculator.getByText('10');
  expect(result).toBeTruthy();
});

test('Using the period will give a float number to use', () => {
  const btnNum = componentCalculator.getByText('5');
  const btnDot = componentCalculator.getByText('.');
  const btnOperator = componentCalculator.getByText('+');
  const btnEqual = componentCalculator.getByText('=');
  fireEvent.click(btnNum);
  fireEvent.click(btnDot);
  fireEvent.click(btnNum);
  fireEvent.click(btnOperator);
  fireEvent.click(btnNum);
  fireEvent.click(btnEqual);

  const result = componentCalculator.getByText('10.5');
  expect(result).toBeTruthy();
});

test('Clicking on +/- will transform the number to negative or positive', () => {
  const btnNum = componentCalculator.getByText('5');
  const btnOperator = componentCalculator.getByText('+/-');
  fireEvent.click(btnNum);
  fireEvent.click(btnNum);
  fireEvent.click(btnOperator);

  const result = componentCalculator.getByText('-55');
  expect(result).toBeTruthy();
});

test('BAD USAGE. Testing on a +/operation, - returning the same number throws an error', () => {
  const t = () => {
    const btnNum = componentCalculator.getByText('5');
    const btnOperator = componentCalculator.getByText('+/-');
    fireEvent.click(btnNum);
    fireEvent.click(btnNum);
    fireEvent.click(btnOperator);

    const result = componentCalculator.getByText('55');
    return result;
  };
  expect(t).toThrow(Error);
});

test('BAD USAGE. Testing on a + operation, returning the wrong number throws an error', () => {
  const t = () => {
    const btnNum = componentCalculator.getByText('5');
    const btnOperator = componentCalculator.getByText('+');
    const btnEqual = componentCalculator.getByText('=');
    fireEvent.click(btnNum);
    fireEvent.click(btnOperator);
    fireEvent.click(btnNum);
    fireEvent.click(btnEqual);

    const result = componentCalculator.getByText('100');
    return result;
  };
  expect(t).toThrow(Error);
});

test('BAD USAGE. Testing on a - operation, returning the wrong number throws an error', () => {
  const t = () => {
    const btnNumOne = componentCalculator.getByText('5');
    const btnNumTwo = componentCalculator.getByText('3');
    const btnOperator = componentCalculator.getByText('-');
    const btnEqual = componentCalculator.getByText('=');
    fireEvent.click(btnNumOne);
    fireEvent.click(btnNumOne);
    fireEvent.click(btnOperator);
    fireEvent.click(btnNumTwo);
    fireEvent.click(btnEqual);

    const result = componentCalculator.getByText('-52');
    return result;
  };
  expect(t).toThrow(Error);
});

test('BAD USAGE. Testing on a X operation, returning the wrong number throws an error', () => {
  const t = () => {
    const btnNumOne = componentCalculator.getByText('5');
    const btnNumTwo = componentCalculator.getByText('3');
    const btnOperator = componentCalculator.getByText('X');
    const btnEqual = componentCalculator.getByText('=');
    fireEvent.click(btnNumOne);
    fireEvent.click(btnOperator);
    fireEvent.click(btnNumTwo);
    fireEvent.click(btnEqual);

    const result = componentCalculator.getByText('53');
    return result;
  };
  expect(t).toThrow(Error);
});

test('BAD USAGE. Testing on a รท operation, returning the wrong number throws an error', () => {
  const t = () => {
    const btnNumOne = componentCalculator.getByText('5');
    const btnNumTwo = componentCalculator.getByText('4');
    const btnOperator = componentCalculator.getByText('รท');
    const btnEqual = componentCalculator.getByText('=');
    fireEvent.click(btnNumOne);
    fireEvent.click(btnNumOne);
    fireEvent.click(btnOperator);
    fireEvent.click(btnNumTwo);
    fireEvent.click(btnEqual);

    const result = componentCalculator.getByText('54');
    return result;
  };
  expect(t).toThrow(Error);
});

test('BAD USAGE. Testing on a % operation, returning the wrong number throws an error', () => {
  const t = () => {
    const btnNumOne = componentCalculator.getByText('1');
    fireEvent.click(btnNumOne);
    const btnNumTwo = componentCalculator.getByText('0');
    const btnOperator = componentCalculator.getByText('%');
    const btnEqual = componentCalculator.getByText('=');
    fireEvent.click(btnNumOne);
    fireEvent.click(btnNumTwo);
    fireEvent.click(btnOperator);
    fireEvent.click(btnNumOne);
    fireEvent.click(btnNumTwo);
    fireEvent.click(btnNumTwo);
    fireEvent.click(btnEqual);

    const result = componentCalculator.getByText('20');
    return result;
  };
  expect(t).toThrow(Error);
});
