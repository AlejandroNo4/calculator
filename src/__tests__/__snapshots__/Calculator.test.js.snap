// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component to Match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="main-container"
        >
          <h1
            class="title"
          >
            Let’s do some math!
          </h1>
          <div
            class="calculatorContainer"
          >
            <div
              class="display"
            >
              0
            </div>
            <div
              class="panel-container"
            >
              <div>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    AC
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    +/-
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    %
                  </span>
                </button>
                <button
                  class="btn"
                  style="background: orange; width: 25%;"
                  type="button"
                >
                  <span>
                    ÷
                  </span>
                </button>
              </div>
              <div>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    7
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    8
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    9
                  </span>
                </button>
                <button
                  class="btn"
                  style="background: orange; width: 25%;"
                  type="button"
                >
                  <span>
                    X
                  </span>
                </button>
              </div>
              <div>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    4
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    5
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    6
                  </span>
                </button>
                <button
                  class="btn"
                  style="background: orange; width: 25%;"
                  type="button"
                >
                  <span>
                    -
                  </span>
                </button>
              </div>
              <div>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    1
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    2
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    3
                  </span>
                </button>
                <button
                  class="btn"
                  style="background: orange; width: 25%;"
                  type="button"
                >
                  <span>
                    +
                  </span>
                </button>
              </div>
              <div>
                <button
                  class="btn"
                  style="width: 50%;"
                  type="button"
                >
                  <span>
                    0
                  </span>
                </button>
                <button
                  class="btn"
                  style="width: 25%;"
                  type="button"
                >
                  <span>
                    .
                  </span>
                </button>
                <button
                  class="btn"
                  style="background: orange; width: 25%;"
                  type="button"
                >
                  <span>
                    =
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="main-container"
      >
        <h1
          class="title"
        >
          Let’s do some math!
        </h1>
        <div
          class="calculatorContainer"
        >
          <div
            class="display"
          >
            0
          </div>
          <div
            class="panel-container"
          >
            <div>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  AC
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  +/-
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  %
                </span>
              </button>
              <button
                class="btn"
                style="background: orange; width: 25%;"
                type="button"
              >
                <span>
                  ÷
                </span>
              </button>
            </div>
            <div>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  7
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  8
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  9
                </span>
              </button>
              <button
                class="btn"
                style="background: orange; width: 25%;"
                type="button"
              >
                <span>
                  X
                </span>
              </button>
            </div>
            <div>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  4
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  5
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  6
                </span>
              </button>
              <button
                class="btn"
                style="background: orange; width: 25%;"
                type="button"
              >
                <span>
                  -
                </span>
              </button>
            </div>
            <div>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  1
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  2
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  3
                </span>
              </button>
              <button
                class="btn"
                style="background: orange; width: 25%;"
                type="button"
              >
                <span>
                  +
                </span>
              </button>
            </div>
            <div>
              <button
                class="btn"
                style="width: 50%;"
                type="button"
              >
                <span>
                  0
                </span>
              </button>
              <button
                class="btn"
                style="width: 25%;"
                type="button"
              >
                <span>
                  .
                </span>
              </button>
              <button
                class="btn"
                style="background: orange; width: 25%;"
                type="button"
              >
                <span>
                  =
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
